File: rubygems/pray-cli/lib/pray/terminal.rb
# frozen_string_literal: true

module Pray
  module Terminal
    module_function

    def color_enabled?
      return false if env_truthy?("PRAY_NO_COLOR") || no_color_requested?

      term = ENV.fetch("TERM", nil)
      term.nil? || term != "dumb"
    end

    def no_color_requested?
      value = ENV.fetch("NO_COLOR", nil)
      !value.nil? && !value.empty?
    end

    def no_input_requested?
      env_truthy?("PRAY_NO_INPUT")
    end

    def env_truthy?(name)
      value = ENV.fetch(name, nil)
      return false if value.nil?

      %w[1 true yes on].include?(value.strip.downcase)
    end
  end
end

File: npmjs/pray-cli/src/terminal.ts
function envTruthy(name: string): boolean {
  const value = process.env[name];
  if (!value) {
    return false;
  }
  const normalized = value.trim().toLowerCase();
  return normalized === "1" || normalized === "true" || normalized === "yes" || normalized === "on";
}

export function noColorRequested(): boolean {
  const value = process.env.NO_COLOR;
  return value !== undefined && value.length > 0;
}

export function colorEnabled(): boolean {
  if (envTruthy("PRAY_NO_COLOR") || noColorRequested()) {
    return false;
  }
  const term = process.env.TERM;
  return term === undefined || term !== "dumb";
}

export function noInputRequested(): boolean {
  return envTruthy("PRAY_NO_INPUT");
}
